src/api_server.cc

```diff
@@ -88,4 +94,32 @@ MetadataApiServer::~MetadataApiServer() {
   }
 }
 
+void MetadataApiServer::HandleMonitoredResource(
+    const HttpServer::request& request,
+    std::shared_ptr<HttpServer::connection> conn) {
+  // The format for the local metadata API request is:
+  //   {host}:{port}/monitoredResource/{id}
+  static const std::string kPrefix = "/monitoredResource/";;
+  const std::string id = request.destination.substr(kPrefix.size());
+  if (config_.VerboseLogging()) {
+    LOG(INFO) << "Handler called for " << id;
+  }
+  try {
+    const MonitoredResource& resource = store_.LookupResource(id);
+    if (config_.VerboseLogging()) {
+      LOG(INFO) << "Found resource for " << id << ": " << resource;
+    }
+    conn->set_status(HttpServer::connection::ok);
```

Comment:
Should you rebase off of master with the latest header updates and 404 response?

Reply:
Yep, done.